packages/dds/merge-tree/src/client.ts

```diff
@@ -257,15 +257,13 @@ export class Client extends TypedEventEmitter<IClientEvents> {
 	 * @param pos - The position to insert the segment at
 	 * @param segment - The segment to insert
 	 */
-	public insertSegmentLocal(pos: number, segment: ISegment): IMergeTreeInsertMsg | undefined {
+	public insertSegmentLocal(pos: number, segment: ISegment): IMergeTreeInsertMsg {
 		if (segment.cachedLength <= 0) {
-			return undefined;
+			throw new UsageError("Cannot insert a 0 length segment.");
```

Comment:
we should add a changeset calling out this change and document expectation that `text` is non-empty on relevant `SharedString` methods. Wouldn't be surprised if users expected this to not throw.

Reply:
I'm actually debating if we should throw or not, as it did impact our tests. I wonder if it useful for callers to know, or if it is just annoying to have to check

Reply:
specifically, would a caller expect some side effect, that wasn't happening due to a zero length insert?

Reply:
The only benefit to our code is really the fuzz test change, as we never want to insert empty in fuzz due to it being a no-op

Reply:
IMO I mildly prefer the older behavior here, yeah.

Reply:
i think i agree. i've rolled back that particular change, but kept some of the test, and less obvious failures.